src/inspector/models/emptystaves/emptystavesvisiblitysettingsmodel.cpp

```diff
@@ -43,6 +45,37 @@ bool EmptyStavesVisibilitySettingsModel::isEmpty() const
     return !selection || !selection->isRange();
 }
 
+void EmptyStavesVisibilitySettingsModel::loadProperties()
+{
+    updateCanHideEmptyStavesInSelection();
+    updateCanShowAllEmptyStaves();
+    updateCanResetEmptyStavesVisibility();
+}
+
+void EmptyStavesVisibilitySettingsModel::onCurrentNotationChanged()
+{
+    INotationPtr notation = currentNotation();
+    if (!notation) {
+        return;
+    }
+
+    notation->undoStack()->changesChannel().onReceive(this, [this](const ScoreChanges& changes) {
+        if (changes.isTextEditing) {
+            return;
+        }
+
+        if (changes.changedPropertyIdSet.empty() && changes.changedStyleIdSet.empty()) {
+            // In this specific case, it's not done by InspectorListModel::listenScoreChanges()
```

Comment:
Maybe we're missing some info in ScoreChanges? Or maybe we can add a new property to AbstractInspectorModel that tells InspectorListModel whether to update this model even if the sets are empty

Reply:
The changes that are relevant here don't go through Sid/Pid, because `MStaff` isn't even an `EngravingObject`. (Changing that is too complicated at this point.)

> maybe we can add a new property to AbstractInspectorModel that tells

Think I'll go with this.